Docs/ArchitectureDecisions/004_observability.md

```diff
@@ -0,0 +1,104 @@
+
+> Note: IoT Hub comes with [curated workbooks](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-explore-curated-visualizations?view=iotedge-2020-11&tabs=devices%2Cmessaging) and [predefined queries for alerts](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-create-alerts?view=iotedge-2020-11) based on built-in Prometheus-format metrics that are delivered by the metrics collector module to LogAnalytics.
+
+#### Custom metrics/events
+
+| Name                       | Description                                                  | Source | Namespace | Dimensions                 |
+| -------------------------- | ------------------------------------------------------------ | ------ | --------- | -------------------------- |
+| MessageDeliveryLatency     | Time from when we received the message from the concentrator until we are done processing it - including downstream send. | LNS    | LoRaWan   | Gateway Id                 |
+| RxWndRate                  | Number of times we hit the different receive windows.        | LNS    | LoRaWan   | Gateway Id, Receive Window |
+| RxWndMiss                  | Number of missed on downstream windows                       | LNS    | LoRaWan   | Gateway Id                 |
+| DeviceCacheHit             | Number of device cache hit                                   | LNS    | LoRaWan   | Gateway Id                 |
+| DeviceLoadRequests         | Number of device load requests                               | LNS    | LoRaWan   | Gateway Id                 |
+| JoinRequests               | Number of join requests                                      | LNS    | LoRaWan   | Gateway Id                 |
+| D2CMessagesReceived        | Number of messages received from device                      | LNS    | LoRaWan   | Gateway Id, Device Id      |
+| D2CMessagesDelivered       | Number of messages sent to upstream                          | LNS    | LoRaWan   | Gateway Id, Device Id, To  |
+| D2CMessagesError           | Number of errors in sending messages to upstream             | LNS    | LoRaWan   | Gateway Id, Device Id, To  |
+| D2CMessagesProcessingError | Number of errors processing (decoding, decrypting) messages  | LNS    | LoRaWan   | Gateway Id, Device Id, To  |
+| D2CMessageSize             | Message size in bytes received from device                   | LNS    | LoRaWan   | Gateway Id, Device Id      |
+| D2CMessageSizeUpstream     | Message size in bytes sent upstream                          | LNS    | LoRaWan   | Gateway Id, Device Id, To  |
+
```

Comment:
What about C2D related metrics? Is there a plan for those?

Reply:
Added the comment that also for C2D we should add these metrics